net/http/src/THttpLongPollEngine.cxx

```diff
@@ -48,12 +57,73 @@ void THttpLongPollEngine::ClearHandle()
    }
 }
 
+//////////////////////////////////////////////////////////////////////////
+/// Create raw buffer which should be send as reply
+/// For the raw mode all information must be send via binary response
+
+void *THttpLongPollEngine::MakeBuffer(const void *buf, int &len, const char *hdr)
+{
+   if (!fRaw) {
+      void *res = malloc(len);
+      memcpy(res, buf, len);
+      return res;
+   }
+
+   int hdrlen = hdr ? strlen(hdr) : 0;
+   std::string hdrstr = "bin:";
+   hdrstr.append(std::to_string(hdrlen));
+
+   while ((hdrstr.length() + 1 + hdrlen) % 8 != 0)
+      hdrstr.append(" ");
+   hdrstr.append(":");
+   if (hdrlen > 0)
+      hdrstr.append(hdr);
+
+   void *res = malloc(hdrstr.length() + len);
+   memcpy(res, hdrstr.c_str(), hdrstr.length());
+   memcpy((char *)res + hdrstr.length(), buf, len);
+   len += hdrstr.length();
+   return res;
+}
+
 //////////////////////////////////////////////////////////////////////////
 /// Send binary data via connection - not supported
 
-void THttpLongPollEngine::Send(const void * /*buf*/, int /*len*/)
+void THttpLongPollEngine::Send(const void *buf, int len)
+{
+   void *buf2 = MakeBuffer(buf, len);
+
+   if (fPoll) {
+      fPoll->SetContentType("application/x-binary");
+      fPoll->SetBinData(buf2, len);
```

Comment:
Shouldn't we at least also have an interface `SetXBinary()` that sets the content type and the data, and use that instead?

And all of the data interfaces should take `vector<char>&&` if they are a data sink (i.e. if that buffer is to be destroyed afterwards).